flink-runtime/src/test/java/org/apache/flink/runtime/concurrent/ScheduledFutureAdapterTest.java

```diff
@@ -0,0 +1,139 @@
+ */
+public class ScheduledFutureAdapterTest {
+
+	private ScheduledFutureAdapter objectUnderTest;
+	private Future<Integer> innerDelegate;
+	private AtomicBoolean resultCancel;
+	private AtomicBoolean resultIsCancelled;
+	private AtomicBoolean resultIsDone;
+
+	@Before
+	public void before() throws Exception {
+		this.resultCancel = new AtomicBoolean(false);
+		this.resultIsCancelled = new AtomicBoolean(false);
+		this.resultIsDone = new AtomicBoolean(false);
+
+		this.innerDelegate = spy(new Future<Integer>() {
+			@Override
+			public boolean cancel(boolean mayInterruptIfRunning) {
+				return resultCancel.get();
+			}
+
+			@Override
+			public boolean isCancelled() {
+				return resultIsCancelled.get();
+			}
+
+			@Override
+			public boolean isDone() {
+				return resultIsDone.get();
+			}
+
+			@Override
+			public Integer get() {
+				return 4711;
+			}
+
+			@Override
+			public Integer get(long timeout, TimeUnit unit) {
+				return 4711;
+			}
+		});
+
+		this.objectUnderTest = new ScheduledFutureAdapter<>(innerDelegate, 4200L, TimeUnit.MILLISECONDS);
+	}
+
+	@Test
+	public void testForwardedMethods() throws Exception {
+
+		Assert.assertEquals(4711, objectUnderTest.get());
+		Mockito.verify(innerDelegate).get();
+
+		Assert.assertEquals(4711, objectUnderTest.get(42L, TimeUnit.SECONDS));
+		Mockito.verify(innerDelegate).get(42L, TimeUnit.SECONDS);
+
+		Assert.assertEquals(resultCancel.get(), objectUnderTest.cancel(true));
+		Mockito.verify(innerDelegate, Mockito.times(1)).cancel(true);
+
+		resultCancel.set(!resultCancel.get());
+		Assert.assertEquals(resultCancel.get(), objectUnderTest.cancel(true));
+		Mockito.verify(innerDelegate, Mockito.times(2)).cancel(true);
+
+		Assert.assertEquals(resultIsCancelled.get(), objectUnderTest.isCancelled());
+		Mockito.verify(innerDelegate, Mockito.times(1)).isCancelled();
+
+		resultIsCancelled.set(!resultIsCancelled.get());
+		Assert.assertEquals(resultIsCancelled.get(), objectUnderTest.isCancelled());
+		Mockito.verify(innerDelegate, Mockito.times(2)).isCancelled();
+
+		Assert.assertEquals(resultIsDone.get(), objectUnderTest.isDone());
+		Mockito.verify(innerDelegate, Mockito.times(1)).isDone();
+
+		resultIsDone.set(!resultIsDone.get());
+		Assert.assertEquals(resultIsDone.get(), objectUnderTest.isDone());
+		Mockito.verify(innerDelegate, Mockito.times(2)).isDone();
+	}
+
+	@Test
+	public void testDelay() {
+
+		Assert.assertEquals(4200L, objectUnderTest.getDelay(TimeUnit.MILLISECONDS));
+		Assert.assertEquals(4L, objectUnderTest.getDelay(TimeUnit.SECONDS));
+
+		final AtomicLong delayUnits = new AtomicLong();
+		Delayed delayed = new Delayed() {
+			@Override
+			public long getDelay(TimeUnit unit) {
+				return delayUnits.get();
+			}
+
+			@Override
+			public int compareTo(Delayed o) {
+				throw new UnsupportedOperationException();
+			}
+		};
+
+		delayUnits.set(objectUnderTest.getDelay(TimeUnit.MILLISECONDS));
+		Assert.assertEquals(0, objectUnderTest.compareTo(delayed));
```

Comment:
wouldn't `a.compareTo(b)==0` also imply `a.equals(b) == true`?

Reply:
See our discussion above, I think it is not quiet that easy here.